bin/sl-deploy.js

```diff
@@ -93,10 +107,27 @@ if (!local)
       return exit(err);
     }
     debug('Connecting to %s via %s', baseURL, url);
-    deploy(process.cwd(), url, branchOrPack, config, exit);
+    var client = new MeshClient(url);
+    client.serviceFindOrCreate(serviceName, 1, function(err, service) {
+      if (err) {
+        if (err.statusCode === 401) {
+          console.error(
+            'Cannot access remote. If authentication is required,' +
+            ' credentials should be given in the URL.');
+        } else {
+          console.error('Error connecting to server:', err);
+        }
+        return exit(err);
+      }
+      deploy(
+        process.cwd(), service.getDeployEndpoint(),
+        branchOrPack,
+        exit
+      );
+    });
   });
 else
-  deploy.local(baseURL, branchOrPack, config, exit);
+  deploy.local(baseURL, branchOrPack, exit);
```

Comment:
arc doesn't use this, right? because this changes the api. the package.json didn't have the major bumped below, I think it should.